WebKit_iosmac builds fail with ASan enabled

ddkilzer opened the pull request:
#### 58784ce9f87d5c3e93834da2fa49b9aa3dd0b94b
<pre>
WebKit_iosmac builds fail with ASan enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=252494">https://bugs.webkit.org/show_bug.cgi?id=252494</a>
&lt;rdar://105597825&gt;

Reviewed by Alexey Proskuryakov.

This adds #if/#endif macros to guard code that uses identifiers
that are not available on the iosmac platform.

* Source/WebKit/Scripts/webkit/messages.py:
(conditions_for_header):
- Add conditions for identifier headers.  Because these
  identifiers have a one-to-one relationship with the headers
  where they are defined, this was the simplest way to add
  support for conditionalizing the identifiers (in addition to
  their headers).
(generate_message_argument_description_implementation):
- Check for conditions when emitting code that references an
  identifier, and output #if/#endif macros to guard it when
  needed.
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
- Regenerate test file which now includes additional #if/#endif
  guards for these identifiers and their headers.

Canonical link: <a href="https://commits.webkit.org/260461@main">https://commits.webkit.org/260461@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b88a56de62ac83c7c76d19d63d46949bc6f8a4e1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108380 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17476 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41245 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117497 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/116865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/18862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8756 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100605 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114147 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/18862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97413 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42142 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/111909 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/18862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29058 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10308 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30402 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11051 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7308 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16458 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49998 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12637 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3930 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Not the code that I'm particularly familiar with, but seems super reasonable.

A little surprised that new code to check for conditions was needed, as the generated file already has a lot of similar conditions, but that's presumably saying more about how the script is designed than about the PR.

Commit:
WebKit_iosmac builds fail with ASan enabled
https://bugs.webkit.org/show_bug.cgi?id=252494
<rdar://105597825>

Reviewed by Alexey Proskuryakov.

This adds #if/#endif macros to guard code that uses identifiers
that are not available on the iosmac platform.

* Source/WebKit/Scripts/webkit/messages.py:
(conditions_for_header):
- Add conditions for identifier headers.  Because these
  identifiers have a one-to-one relationship with the headers
  where they are defined, this was the simplest way to add
  support for conditionalizing the identifiers (in addition to
  their headers).
(generate_message_argument_description_implementation):
- Check for conditions when emitting code that references an
  identifier, and output #if/#endif macros to guard it when
  needed.
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
- Regenerate test file which now includes additional #if/#endif
  guards for these identifiers and their headers.

Canonical link: https://commits.webkit.org/260461@main
